<!doctype html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport"
          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>
        1.await只能放在async函数里面
        2.await后面可以是promise对象 也可以是数字 字符串
        3.async函数返回的是promise对象
        4.只要await语句后面Promise状态变成reject，那么整个async函数都会中断执行
        5.可以使用try catch解决中断问题以防影响后面的代码
    </title>
</head>
<body>
<script>
    // 2s 之后返回双倍的值
    function doubleAfter2seconds(num) {
        return new Promise((resolve, reject) => {
            setTimeout(() => {
                resolve(2 * num)
            }, 2000);
        } )
    }
    async function testResult() {
        let result = await doubleAfter2seconds(30);
        console.log(result);//2s后打印出来60
        let   second = await doubleAfter2seconds(50);
        console.log(second)//4s后打印出100
        let   third = await doubleAfter2seconds(30);
        console.log(third)//6s后打印出来60
    }
    testResult() //60

</script>
</body>
</html>